File: Practice/실습/201104/5521.cpp
#include <iostream>
#include <vector>
#include <queue>

#define FRIENDS 501

using namespace std;

int main(int argc, char** argv) {
    int T;
    cin >> T;
    for (int t = 1; t <= T; t++) {
        int N, M;
        cin >> N >> M;
        int start, end;
        vector<int> G[FRIENDS];
        for (int i = 0; i < M; i++) {
            cin >> start >> end;
            G[start].push_back(end);
            G[end].push_back(start);
        }
        int result = 0;
        queue<int> q;
        int check[FRIENDS] = {0,};
        q.push(1);
        check[1] = 1;
        int v;
        while (!q.empty()) {
            v = q.front();
            q.pop();
            if (check[v] > 3) {
                break;
            }
            for (int i = 0; i < G[v].size(); i++) {
                int w = G[v][i];
                if (!check[w]) {
                    q.push(w);
                    check[w] = check[v] + 1;
                    if (check[w] < 4) {
                        result++;
                    }
                }
            }
        }
        cout << '#' << t << ' ' << result << '\n';
    }
    return 0;
}

File: Practice/실습/201104/5521.py
T = int(input())
for t in range(1, T+1):
    N, M = map(int, input().split())
    rel = [list(map(int, input().split())) for _ in range(M)]
    G = [[0] * (N+1) for _ in range(N+1)]
    for s, e in rel:
        G[s][e] = G[e][s] = 1
    result = 0
    queue = []
    visited = [0] * (N+1)
    queue.append(1)
    visited[1] = 1
    while queue:
        v = queue.pop(0)
        if visited[v] > 3:
            break
        for w in range(N+1):
            if G[v][w] and not visited[w]:
                queue.append(w)
                visited[w] = visited[v] + 1
                if visited[w] < 4:
                    result += 1
    print('#{} {}'.format(t, result))
